zk_toolbox/crates/common/src/withdraw.rs

```diff
@@ -1,18 +1,68 @@
 use async_trait::async_trait;
+// use serde_json::Value;
 use ethers::{
-    providers::ProviderError,
-    types::{Address, Bytes, H160, H256, U64},
+    providers::{JsonRpcClient, Provider, ProviderError},
+    types::Bytes,
+    types::{Address, TransactionReceipt, TxHash, H160, H256, U64},
 };
 use serde::{Deserialize, Serialize};
+use serde_json::json;
 use zksync_system_constants::L1_MESSENGER_ADDRESS;
-use zksync_types::{
-    api::{L2ToL1Log, L2ToL1LogProof, Log},
-    ethabi,
-};
-use zksync_web3_decl::{
-    client::{Client, L2},
-    namespaces::{EthNamespaceClient, ZksNamespaceClient},
-};
+use zksync_types::ethabi;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Proof {
+    pub id: u64,
+    #[serde(rename = "proof")]
+    pub merkle_proof: Vec<H256>,
+    pub root: Bytes,
+}
+
+/// Represents a L2 to L1 transaction log.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct L2ToL1Log {
+    /// The block number.
```

Comment:
I may be wrong, but it seems to repeat similar issues to the PR where the l2-erc20 test was added, could you please look into my comments there and how Kalman fixed them